### PR TITLE
detach volume after unmount

### PR DESCRIPTION
**Pull request by calvinyv**

### CODE REVIEW

File: pkg/volume/qingcloud/flex_volume_plugin.go [hunk 1]
```diff
@@ -140,14 +140,28 @@ func (p *flexVolumePlugin) MountDevice(dir, _ string, options flex.VolumeOptions
...
+func (p *flexVolumePlugin) UnmountDevice(dir string) flex.VolumeResult {
 	glog.V(4).Infof("UnmountDevice %v", dir)
 	exec := mount.NewOsExec()
 	mounter := &mount.SafeFormatAndMount{Interface: mount.New(""), Exec: exec}
 	err := mounter.Unmount(dir)
 	if err != nil {
 		return flex.NewVolumeError(err.Error())
 	}
+	volumeName := getVolumeName(dir)
+	volumeID, node , err:= p.manager.GetVolumeInfoByName(volumeName)
```
Comment (chai2010) [inline pkg/volume/qingcloud/flex_volume_plugin.go]: please run `go fmt ./...`

Comment (calvinyv) [reply pkg/volume/qingcloud/flex_volume_plugin.go]: done

File: pkg/volume/qingcloud/util_test.go [hunk 2]
```diff
@@ -10,6 +10,13 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 )
 
+func TestGetVolumeName(t *testing.T) {
+	path := "data/var/lib/kubelet/plugins/kubernetes.io/flexvolume/qingcloud/flex-volume/mounts/pvc-ab52a0f1-fa62-11e7-806e-52541fed8204"
+	volName := getVolumeName(path)
+	if volName != "pvc-ab52a0f1-fa62-11e7-806e-52541fed8204" {
+		t.Errorf("case GetVolumeName: expect %v but get %v", "pvc-ab52a0f1-fa62-11e7-806e-52541fed8204", volName)
```
Comment (chai2010) [inline pkg/volume/qingcloud/util_test.go]: `s/get/got/` ?

File: pkg/volume/qingcloud/volume_manager.go [hunk 3]
```diff
@@ -69,11 +69,12 @@ type VolumeManager interface {
 	UpdateVolume(volumeID, volumeName string) error
 
 	//GetVolumeIDByName
-	GetVolumeIDByName(volumeName string) (string, error)
+	GetVolumeInfoByName(volumeName string) (string, string, error)
```
Comment (chai2010) [inline pkg/volume/qingcloud/volume_manager.go]: please name the returned value

File: pkg/volume/qingcloud/volume_manager.go [hunk 4]
```diff
@@ -69,11 +69,12 @@ type VolumeManager interface {
 	UpdateVolume(volumeID, volumeName string) error
 
 	//GetVolumeIDByName
```
Comment (chai2010) [inline pkg/volume/qingcloud/volume_manager.go]: should contain a space between `//` and comments

**Review by chai2010 — APPROVED** [review]
LGTM

**Pull request merged**